Add "Development Tutorial" documentation (#5324)
antrea-io/antrea: docs/contributors/dev-tutorial.md

```go
func (c *Controller) addTraceflow(obj interface{}) {
    tf := obj.(*crdv1alpha1.Traceflow)
    klog.InfoS("Processing Traceflow ADD event", "uid", tf.Spec.UID)
    // ...
}
```

Likewise, modify the `addTraceflow` function in
`pkg/controller/traceflow/controller.go`.

After making the changes, rebuild the Docker image:

```bash
make antrea-agent antrea-controller antrea-cni antctl-release
NO_PULL=1 make ubuntu
```

The built Docker image will not be automatically loaded into the Kind cluster.
You need to run the following commands to load the Docker image into the Kind
cluster and force Antrea to use the new image:

```bash
kind load docker-image antrea/antrea-ubuntu:latest
kubectl -n kube-system rollout restart deployment/antrea-controller daemonset/antrea-agent
```

### Initial testing

Before moving on, run the unit tests and e2e tests related to the changes to
check for any errors:

```bash
go test -v ./pkg/agent/controller/traceflow ./pkg/controller/traceflow
ci/kind/test-e2e-kind.sh --run TestTraceflow --test-only --encap-mode encap
```

The `--encap-mode encap` option is used to run only the `encap` mode e2e tests.
Antrea also supports `noEncap` and `hybrid` modes.

> Next, manually test whether the new feature works. First, create a new
> Traceflow:

Reviewer: ```suggestion
After run above tests successfully, you can create a new Traceflow to verify the change manually to check whether the new feature works:
```

```bash
kubectl apply -f - <<EOF
apiVersion: crd.antrea.io/v1alpha1
kind: Traceflow
metadata:
  name: tf-test
spec:
  liveTraffic: true
  source:
    namespace: default
    pod: nginx1
  destination:
    namespace: default
    pod: nginx2
  uid: foobar
EOF
```

Then, check whether antrea-agent and antrea-controller have logged the
corresponding messages:

For antrea-agent:

```bash
ANT_AGENT=$(kubectl -n kube-system get pods -o wide | grep "agent.*worker " | cut -d ' ' -f 1)
kubectl -n kube-system logs $ANT_AGENT | grep "Processing Traceflow ADD event"
```

For antrea-controller:

```bash
ANT_CONTROLLER=$(kubectl -n kube-system get pods -o wide | grep "antrea-controller" | cut -d ' ' -f 1)
kubectl -n kube-system logs $ANT_CONTROLLER | grep "Processing Traceflow ADD event"
```

> If everything is working correctly, when running the above commands, you should
> see output similar to the following:

Reviewer: ```suggestion
If everything is working correctly, you should
see output similar to the following after run the above commands:
```

```text
I0727 08:05:39.259424       1 traceflow_controller.go:192] "Processing Traceflow ADD event" uid="foobar"
```

### Write unit tests and e2e tests

Since this modification is relatively simple, there is no need to add new unit

> tests or e2e tests. However, if you plan to make significant changes, it's a
> good idea to add relevant unit tests and e2e tests.

Reviewer: ```suggestion
tests or e2e tests. However, if you plan to make significant changes, it's recommended to add relevant unit tests and e2e tests.
```

## Commit the code

> Before committing the code, run the tests one last time locally to ensure there

Reviewer: ```suggestion
Before commit the code, run the tests one last time locally to ensure there
```
